File: server/lib/silver_brain/concept_map/concept.ex
defmodule SilverBrain.ConceptMap.Concept do
  @type t :: %__MODULE__{
          uuid: String.t(),
          name: String.t(),
          content_type: nil | String.t(),
          content: nil | String.t(),
          create_time: DateTime.t(),
          update_time: DateTime.t()
        }

  @derive Jason.Encoder
  defstruct [:uuid, :name, :content_type, :content, :create_time, :update_time]
end

File: server/src/SilverBrain/ConceptMap/Core/Concept.hs
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE RecordWildCards #-}

module SilverBrain.ConceptMap.Core.Concept where

import Data.Aeson (FromJSON, ToJSON)
import Data.Text (Text)
import Data.Time (UTCTime)
import GHC.Generics
import SilverBrain.ConceptMap.Core.Types
import Text.Printf (printf)

data Concept = Concept
  { uuid :: Uuid,
    name :: Uuid,
    contentType :: Uuid,
    content :: Uuid,
    createTime :: UTCTime,
    updateTime :: UTCTime
  }
  deriving (Generic)

instance Show Concept where
  show Concept {..} = printf "#Concept{%s}" name

instance FromJSON Concept

instance ToJSON Concept
